### PR TITLE
Different entity types

### PR DESCRIPTION
**Pull request by tmjo**
Adding ability for different entity types (sensors, switches etc). Moved most of the stuff from sensor.py to a new entity.py, leaving sensor quite small and adding a switch.py for some new sensors. Moving entity setup to const.

Pls see if this makes sense.

### CODE REVIEW

File: custom_components/easee/__init__.py [hunk 1]
```diff
@@ -64,20 +74,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
...
         for circuit in site.get_circuits():
+            _LOGGER.debug("Found circuit: %s %s", circuit.id, circuit["panelName"])
             hass.data[DOMAIN]["circuits"].append(circuit)
             for charger in circuit.get_chargers():
+                _LOGGER.debug("Found charger: %s %s", charger.id, charger.name)
                 hass.data[DOMAIN]["chargers"].append(charger)
+                charger_data = ChargerData(charger, circuit, site)
+                charger_data_list.append(charger_data)
+
+    # config = hass.data[DOMAIN]["config"]
```
Comment (fondberg) [inline custom_components/easee/__init__.py:91]: remove comment

File: custom_components/easee/const.py [hunk 2]
```diff
@@ -1,3 +1,278 @@
+"""Easee Charger constants."""
 DOMAIN = "easee"
 MEASURED_CONSUMPTION_DAYS = "measured_consumption_days"
 VERSION = "0.9.11"
+PLATFORMS = ("sensor", "switch")
+SCAN_INTERVAL_SECONDS = 60
+EASEE_ENTITIES = {
```
Comment (fondberg) [inline custom_components/easee/const.py:7]: good

**Comment by fondberg** [comment]
You wanna try a release yourself? Just merge this and checkout master of this repo and do a `make bump` then check it in and go to releases and create a release out from master.

**Comment by tmjo** [comment]
> You wanna try a release yourself? Just merge this and checkout master of this repo and do a `make bump` then check it in and go to releases and create a release out from master.

Cool, I'll give it a try. But I'll verify that it's working after the update of the site PR first.

**Comment by tmjo** [comment]
Ok, so I rebased and it should be working well. @olalid; would you mind testing this before we merge just to verify I didn't mess up your changes regarding sites? I've tested it and it is ok on my side, but I only got one site. However, I'm to enable and disable it successfully, so I think it is ok. Due to the different entity types, a lot of the logics were moved from sensor.py to entity.py and __init__.py

**Comment by olalid** [comment]
Looks ok to me.
One minor thing though, the links.sh script needs to be updated so that entity.py and switch.py gets links created.

**Comment by tmjo** [comment]
Good catch! Added both files to links.sh. Is it possible to link the whole folder instead of each separate file? Just curious.

**Comment by olalid** [comment]
> Good catch! Added both files to links.sh. Is it possible to link the whole folder instead of each separate file? Just curiousI

If you check the project out somewhere else than in custom_components/easee you could link the right folder to that location instead. So the concept of how the project is intended to be checked out needs to change I guess.

**Comment by tmjo** [comment]
Ok. I'll merge it but wait a bit to release, have a couple of minor adjustments I'll send separately. How should we proceed in general now when the three of us can merge and release? Should we request a thumb up from the trio or at least 2/3 or something before merging and releasing? For everything, or just major/controversial stuff and not bugfix etc? How do you guys prefer it? @fondberg @olalid

**Comment by olalid** [comment]
Well, I guess it may depend on how confident one feels about the change, and it is easy to feel confident about minor things...
So something like you said I guess, bug fixes, go ahead, other stuff, discussion.
Thats my opinion at least.

**Comment by fondberg** [comment]
Hot fixes and small bug fixes can be merged directly.
Everything as a PR and squash merged as one master commit.

Other things as a PR with possibility for feedback from the rest.
Each PR should ideally only contain code changes for one "thing". Not fixed x, added x and z

**Pull request merged**